.github/workflows/plugins.yml

```diff
@@ -1,49 +1,70 @@
 name: plugins
 on:
+# Run CI against all pushes (direct commits) and Pull Requests
   push:
-    paths:
-      - 'plugins/**'
-      - 'tests/**'
-      - 'scripts/**'
-      - '.github/workflows/plugins.yml'
-      - 'docker-compose.yml'
+#    paths:
+#      - 'plugins/**'
+#      - 'tests/**'
+#      - 'scripts/**'
+#      - '.github/workflows/plugins.yml'
+#      - 'docker-compose.yml'
   pull_request:
-    paths:
-      - 'plugins/**'
-      - 'tests/**'
-      - 'scripts/**'
-      - '.github/workflows/plugins.yml'
-      - 'docker-compose.yml'
+#    paths:
+#      - 'plugins/**'
+#      - 'tests/**'
+#      - 'scripts/**'
+#      - '.github/workflows/plugins.yml'
+#      - 'docker-compose.yml'
 
 jobs:
   sanity:
-    name: Sanity (${{ matrix.ansible }})
+    name: Sanity (Ⓐ${{ matrix.ansible }}+py${{ matrix.python }})
     strategy:
       matrix:
         ansible:
-        - stable-2.10
-        - devel
+          # It's important that Sanity is tested against all stable-X.Y branches 
+          # Testing against `devel` may fail as new tests are added.
+        # - stable-2.9 # Only if your collection supports Ansible 2.9
+          - stable-2.10
+          - devel
+        python:
+          - 2.7
+          - 3.7
+          - 3.8
+        exclude:
+          - python: 3.8  # blocked by https://github.com/ansible/ansible/pull/70155
     runs-on: ubuntu-latest
     steps:
 
+      # ansible-test requires the collection to be in a directory in the form
+      # .../ansible_collections/NAMESPACE/COLLECTION_NAME/
+
       - name: Check out code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
           path: ansible_collections/community/zabbix
 
-      - name: Set up Python 3.6
-        uses: actions/setup-python@v1
+      - name: Set up Python ${{ matrix.ansible }}
+        uses: actions/setup-python@v2
         with:
-          python-version: 3.6
+          python-version: ${{ matrix.python }}
 
+      # Install the head of the given branch (devel, stable-2.10)
       - name: Install ansible-base (${{ matrix.ansible }})
         run: pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible }}.tar.gz --disable-pip-version-check
 
+      # run ansible-test sanity inside of Docker.
+      # The docker container has all the pinned dependencies that are required.
+      # Explicity specify the version of Python we want to test
       - name: Run sanity tests
-        run: ansible-test sanity --docker -v --color --python 3.6
+        run: ansible-test sanity --docker -v --color --python ${{ matrix.python }}
+        working-directory: ./ansible_collections/community/zabbix
+
+# FIXME IN TEMPLATE: UNIT TESTS WOULD GO HERE
 
   integration:
     runs-on: ubuntu-latest
+    name: I (${{ matrix.zabbix_container.version}}:${{ matrix.zabbix_container.port }} Ⓐ${{ matrix.ansible }}+py${{ matrix.python }}})
```

Comment:
Port can go away altogether, no need to have it in the name

Reply:
@D3DeFi Thank you, that looks a lot better.